token/cli/src/main.rs

```diff
@@ -3184,6 +3258,43 @@ async fn handle_tx<'a>(
     }
 }
 
+async fn finish_tx<'a>(
+    config: &Config<'a>,
+    rpc_response: &RpcClientResponse,
+    no_wait: bool,
+) -> Result<TransactionReturnData, Error> {
+    match rpc_response {
+        RpcClientResponse::Transaction(transaction) => {
+            Ok(TransactionReturnData::CliSignOnlyData(return_signers_data(
+                &transaction,
+                &ReturnSignersConfig {
+                    dump_transaction_message: config.dump_transaction_message,
+                },
+            )))
+        }
+        RpcClientResponse::Signature(signature) if no_wait => {
+            Ok(TransactionReturnData::CliSignature(CliSignature {
+                signature: signature.to_string(),
+            }))
```

Comment:
At first, I wrote that you should send here, before realizing that the send is already being done in the client -- nice!